mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2197,29 +2284,27 @@ static func::FuncOp createVerifierFunc(ModuleOp module, const KernelIF &kernel,
 
   // obtain function name of the verifier wrapper
   std::string verifyFuncName = "mcpuVerify";
-  if (valElemType.isF32()) {
+  if (valElemType.isa<FloatType>()) {
```

Comment:
Why is this test relaxed, especially since below you assume `valElemType` is `f32`?

Reply:
Not sure where I'm assuming, but I think it's a hangover from when I thought the CPU could handle any float type.  Anyway, undone.